src/main/java/com/example/eventorganizer/task/ContributionService.java

```diff
@@ -0,0 +1,31 @@
+package com.example.eventorganizer.task;
+
+import com.example.eventorganizer.event.Event;
+import com.example.eventorganizer.user.User;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+@Service
+public class ContributionService {
+    public Map<User, BigDecimal> calculateContributions(Event event, User spender, BigDecimal spentMoney){
+        List<User> participants = new ArrayList<>(event.getUsers());
+        participants.remove(spender);
+
+        int numParticipants = participants.size() + 1;
+        if( numParticipants == 1){
+            return Collections.emptyMap();
+        }
+
+        BigDecimal contributionPerPerson = spentMoney.divide(BigDecimal.valueOf(numParticipants),2, RoundingMode.HALF_UP);
+
+        Map<User, BigDecimal> contributions = new HashMap<>();
```

Comment:
You can transform list of users into map using java 8+ 's collection methods. No need to create new one yourself.
https://www.baeldung.com/java-list-to-map#before-java8